playlistmanager.lua

```diff
@@ -300,28 +308,18 @@ function on_loaded()
     mp.set_property("title", settings.title_prefix..strippedname..settings.title_suffix)
   end
 
-  local didload = false
   if settings.loadfiles_on_start and plen == 1 then
     local ext = filename:match("%.([^%.]+)$")
     -- a directory or playlist has been loaded, let's not do anything as mpv will expand it into files
     if ext and filetype_lookup[ext:lower()] then
-      didload = true --save reference for sorting
       msg.info("Loading files from playing files directory")
       playlist()
     end
   end
 
-  --if we promised to sort files on launch do it
-  if promised_sort then
-    promised_sort = false
-    msg.info("Your playlist is sorted before starting playback")
-    if didload then sortplaylist() else sortplaylist(true) end
-  end
-
   --if we promised to listen and sort on playlist size increase do it
   if promised_sort_watch then
     promised_sort_watch = false
-    sort_watching = true
     msg.info("Added files will be automatically sorted")
     mp.observe_property('playlist-count', "number", autosort)
   end
```

Comment:
I think now that this function is removed if the observe_property could be moved out of the onloaded event directly inside this if statement:

```
if settings.sortplaylist_on_file_add then
  promised_sort_watch = true
end
```

Reply:
Or actually maybe the reason it was like this is to avoid a sort on the first load. Need to test it out a bit.

Not sure if people are using these features much anyway.

Reply:
Seems like there is some race condition on this autosort feature. `plen` variable is refreshed before this is ran. It seems unrelated to your PR though.

Reply:
I just thought of a way to fix it, but I haven't pushed it yet. I'll open a new pr for this.